scripts/supported_apis.py

```diff
@@ -0,0 +1,348 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Get supported APIs list by parsing docstrings in the Modin source files.
+
+Example usage:
+python scripts/supported_apis.py
+"""
+
+import pathlib
+import os
+import sys
+
+import ast
+from numpydoc.validate import Docstring
+import pandas
+
+MODIN_PATH = pathlib.Path(__file__).parent.parent.absolute()
+sys.path.insert(0, str(MODIN_PATH))
+
+dataframe_supported_csv_path = (
+    MODIN_PATH / "docs/supported_apis/dataframe_supported.csv"
+)
+series_supported_csv_path = MODIN_PATH / "docs/supported_apis/series_supported.csv"
+io_supported_csv_path = MODIN_PATH / "docs/supported_apis/io_supported.csv"
+utilities_supported_csv_path = (
+    MODIN_PATH / "docs/supported_apis/utilities_supported.csv"
+)
+dataframe_path = pathlib.Path("modin/pandas/dataframe.py")
+series_path = pathlib.Path("modin/pandas/series.py")
+base_path = pathlib.Path("modin/pandas/base.py")
+io_path = pathlib.Path("modin/pandas/io.py")
+utilities_path = pathlib.Path("modin/pandas/general.py")
+
+csv_columns = [
+    "Method Name",
+    "Parameter",
+    "PandasOnRay",
+    "PandasOnDask",
+    "OmniSci",
+    "Notes",
+]
+
+# these utilities just imported from pandas in `modin.pandas.__init__`
+pandas_utilities = [
```

Comment:
I'm wondering can we somehow avoid hardcoding these names since all they are taken from __init__ anyway?

Reply:
It can be tricky, i think - not all the objects imported from pandas in the **init** file are intended to be added in the "general functions" section (`RangeIndex` for example - it is imported in the **init** file but shouldn't to be processed here), so we have to filter this list somehow and it can be hard to make without such hardcoding.